src/words.ts

```diff
@@ -0,0 +1,147 @@
+import {mmod} from 'xen-dev-utils';
+
+type StepVector = Record<string, number>;
+
+function sum(array: number[]) {
+  let sum = 0;
+  array.forEach(x => {
+    sum += x;
+  });
+  return sum;
+}
+
+export function zeroVector(): StepVector {
+  return {};
+}
+
+export function add(v1: StepVector, v2: StepVector): StepVector {
+  // Clone because we don't want to mutate the input.
+  const result = {...v1};
+  for (const [letter, value] of Object.entries(v2)) {
+    // For each entry,
+    // if `result` does not have the entry then add the component from v2 as an entry.
+    if (!(letter in v1)) {
+      result[letter] = value;
+    }
+    // If `result` has a nonzero coefficient for that entry:
+    else {
+      // If the sum at the component is 0, then delete the entry.
+      if (value + v1[letter] === 0) {
+        delete result[letter];
+      } else {
+        // Otherwise update the component to the sum.
+        result[letter] += value;
+      }
+    }
+  }
+  return result;
+}
+
+export function scalarMult(scalar: number, v: StepVector): StepVector {
+  if (scalar === 0) {
+    return zeroVector();
+  } else {
+    // Clone because we don't want to mutate the input.
+    const result = {...v};
+    for (const letter of Object.keys(result)) {
+      result[letter] *= scalar;
+    }
+    return result;
+  }
+}
+
+// Additive inverse of a `StepVector`
```

Comment:
Comments with `//` are invisible to TypeDoc. Try with `/*...*/` and make sure that the result appears in the output when you do `npm run doc`.